start.R

```diff
@@ -38,26 +38,49 @@ if (length(argv) > 0) {
 }
 
 # read in configuration
-cfg <- gms::check_config(cfg, modulepath = NULL)
+cfg <- gms::check_config(cfg, reference_file="config/APT.cfg", modulepath = NULL)
 
 # use cachefolder from configuration file if exists
 if (!is.null(cfg$cachefolder)) {
   madrat::setConfig(cachefolder = cfg$cachefolder)
 }
 
-for (mapping in cfg$mappinglist) {
-
-  # Produce input data for all regionmappings (ignore extramappings_historic)
-  retrieveData(model = "REMIND", regionmapping = mapping[["regionmapping"]],
-               rev = cfg$revision, dev = cfg$dev, cachetype = cfg$cachetype,
-               renv = cfg$renv)
-
-  # Produce historical data for regionmappings and extramappings_historic.
-  # The region hash of the historical data file will concatenated from the 
-  # individual hashes of regionmapping and extramappings_historic.
-  retrieveData(model = "VALIDATIONREMIND",
-               regionmapping = mapping[["regionmapping"]],
-               extramappings = mapping[["extramappings_historic"]],
-               rev = cfg$revision, dev = cfg$dev, cachetype = cfg$cachetype,
-               renv = cfg$renv)
+stoppedWithError <- tryCatch({
+  for (mapping in cfg$mappinglist) {
+    # Produce input data for all regionmappings (ignore extramappings_historic)
+    retrieveData(model = "REMIND", regionmapping = mapping[["regionmapping"]],
+                 rev = cfg$revision, dev = cfg$dev, cachetype = cfg$cachetype,
+                 renv = cfg$renv)
+
+    # Produce historical data for regionmappings and extramappings_historic.
+    # The region hash of the historical data file will concatenated from the 
+    # individual hashes of regionmapping and extramappings_historic.
+    retrieveData(model = "VALIDATIONREMIND",
+                 regionmapping = mapping[["regionmapping"]],
+                 extramappings = mapping[["extramappings_historic"]],
+                 rev = cfg$revision, dev = cfg$dev, cachetype = cfg$cachetype,
+                 renv = cfg$renv)
+  }  
+  FALSE
+}, error = function(error) {
+  print(error)
+  return(TRUE)
+})
+
+# If this is an APT send bot message to mattermost in case the APT produced warnings or errors
```

Comment:
is there a better place for this? For other than APT runs it is just unused code...

Reply:
That' true, but I wouldn't know where to better put it since we only have the one start.R script.